src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/StartRequestHandler.kt

```diff
@@ -84,13 +69,24 @@ class StartRequestHandler(private val config: FrejaEidAuthenticatorPluginConfig,
     }
 
     override fun get(requestModel: RequestModel, response: Response): Optional<AuthenticationResult> =
-            if (authenticatedState.isAuthenticated)
+
+            if (config.qrCodeEnabled())
             {
-                startAuthentication(requestModel)
+                //if qrcode is enabled, throw to go to wait handler
+                // and skip the form entry for email, ssn or phone
+                throw _exceptionFactory.redirectException(getWaitHandlerUrl(),
+                        RedirectStatusCode.MOVED_TEMPORARILY, emptyMap(), false)
             }
             else
```

Comment:
else if ... else

why the nesting of the if else in the outer if?

Reply:
fixed